hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -406,38 +406,54 @@ def setup_occupancy_type(
+                gdf = gdf.loc[gdf["Primary Object Type"] != ""]
 
-            nr_without_landuse = len(gdf.loc[gdf["Primary Object Type"].isna()].index)
-            if nr_without_landuse > 0:
-                self.logger.warning(
-                    f"{nr_without_landuse} objects were not overlapping with the "
-                    "land use data and will be removed from the exposure data."
-                )
-            gdf = gdf.loc[gdf["Primary Object Type"].notna()]
+                nr_without_landuse = len(gdf.loc[gdf["Primary Object Type"].isna()].index)
+                if nr_without_landuse > 0:
+                    self.logger.warning(
+                        f"{nr_without_landuse} objects were not overlapping with the "
+                        "land use data and will be removed from the exposure data."
+                    )
+                gdf = gdf.loc[gdf["Primary Object Type"].notna()]
 
-            # Update the exposure geoms
-            self.exposure_geoms[0] = gdf[["Object ID", "geometry"]]
 
             # Remove the geometry column from the exposure database
             del gdf["geometry"]
             # Update the exposure database
             if type_add in self.exposure_db:
-                gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
-                self.exposure_db = pd.merge(
-                    self.exposure_db, gdf, on="Object ID", how="left"
-                )
-                self.exposure_db = self._set_values_from_other_column(
-                    self.exposure_db, "Primary Object Type", "pot"
-                )
+                if "Primary Object Type" in gdf.columns:
+                    gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
+                    self.exposure_db = pd.merge(
+                        self.exposure_db, gdf, on="Object ID", how="left"
+                    )
+                    self.exposure_db = self._set_values_from_other_column(
+                        self.exposure_db, "Primary Object Type", "pot"
+                    )
+                    
+                    # Also rpelace Secondary Object with new classification to assign correct damage curves
+                    self.exposure_db = pd.merge(
+                        self.exposure_db, gdf, on="Object ID", how="left"
+                    )
+                    self.exposure_db = self._set_values_from_other_column(
+                        self.exposure_db, "Secondary Object Type", "pot"
+                    )
+                elif "Secondary Object Type" in gdf.columns:
+                    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
+                    self.exposure_db = pd.merge(
+                        self.exposure_db, gdf, on="Object ID", how="left"
+                    )
+                    self.exposure_db = self._set_values_from_other_column(
+                        self.exposure_db, "Secondary Object Type", "pot"
+                    )
```

Comment:
To make it more intuitive to read and understand, it is good practice to handle exceptions/edge cases first, and then after handling all exceptions, handle the function's responsibility.


```
....
# Update the exposure database
if type_add not in self.exposure_db:
   self.logger.warning(
    "NotImplemented the spatial join of the exposure data with the "
    "occupancy map the for multiple exposure geoms"
    )
    NotImplemented

if ("Primary Object type" not in gdf.columns) and ("Secondary Object Type" not in gdf.columns):
       self.exposure_db = gdf.copy()
       return

if "Primary Object Type" in gdf.columns:
    gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Primary Object Type", "pot"
    )
    
    # Also rpelace Secondary Object with new classification to assign correct damage curves
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Secondary Object Type", "pot"
    )
elif "Secondary Object Type" in gdf.columns:
    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Secondary Object Type", "pot"
    )
```

Reply:
Additionally, only one of these blocks will be executed in a function call. 
`elif <expr>` will only be evaluated if all previous if & elif statements in the block were false.
Use 2 `if` statements if you want to do the check for both

```
if "Primary Object Type" in gdf.columns:
    ....
elif "Secondary Object Type" in gdf.columns:
    .....
```

Reply:
I think the logger warning refers to the if-statement some more lines above:
if len(self.exposure_geoms) == 1. But following your example i could update it like this;

`            if type_add not in self.exposure_db:
                    self.exposure_db = gdf.copy()
             else
                if "Primary Object Type" in gdf.columns:
                    gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Primary Object Type", "pot"
                    )
                    
                    # Also rpelace Secondary Object with new classification to assign correct damage curves
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Secondary Object Type", "pot"
                    )
                elif "Secondary Object Type" in gdf.columns:
                    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Secondary Object Type", "pot"
                    )
            `

Reply:
> Additionally, only one of these blocks will be executed in a function call. `elif <expr>` will only be evaluated if all previous if & elif statements in the block were false. Use 2 `if` statements if you want to do the check for both
> 
> ```
> if "Primary Object Type" in gdf.columns:
>     ....
> elif "Secondary Object Type" in gdf.columns:
>     .....
> ```

 In this case I want to only run one block and not no other if a condition was True beforehand

Reply:
> I think the logger warning refers to the if-statement some more lines above: if len(self.exposure_geoms) == 1. But following your example i could update it like this;
> 
> ```
>        `if type_add not in self.exposure_db:
>                 self.exposure_db = gdf.copy()
>         else:
>              ...
Oh oops, you are right